inference-engine/thirdparty/clDNN/src/gpu/kernels_cache.cpp

```diff
@@ -412,9 +434,14 @@ kernels_cache::kernels_map kernels_cache::build_program(const program_code& prog
         }
 
         if (!err_log.empty()) {
-            static const size_t max_msg_length = 128;
+            auto str_max_msg_length = GetStringEnv("CL_DNN_MAX_OCL_ERROR");
+            size_t max_msg_length = 128;
+            try {
+                max_msg_length = std::stoi(str_max_msg_length);
+            } catch(...) {}
+
             std::string short_err_log(err_log, 0, std::min(err_log.length(), max_msg_length));
-            throw std::runtime_error("Program build failed:\n" + std::move(short_err_log));
+            throw std::runtime_error("Program build failed (You may set env variable CL_DNN_MAX_OCL_ERROR):\n" + std::move(short_err_log));
```

Comment:
Is it really useful to get longer error message without the code? I mean we have a possibility to [dump ocl source code](https://github.com/openvinotoolkit/openvino/wiki/GPUPluginDebugUtils#sources-dumps) where compilation errors are added to the end of each file, so you can navigate from error log to the exact code line that causes the error.

Reply:
Agree with your point.

Currently, truncated message misleads developer as it shows irrelevant warning only. That is what I want to fix.

For example, if I mistyped one character in .cl file, execution log shows a warning that is not related to my fix at all. The message does not provide any clue.

I think we have two options to improve developer's experience:
1) If dump-ocl-source is the main(perhaps only) dev flow, we should just not print any ocl compile error message. Truncation may mislead people. Instead, it may just print ```Program build failed. You may enable OCL source dump to see the error log```
2) For developer who may not enable dump-ocl-source, we may support to printing error log in exception message.

How do you think?

Reply:
@isanghao we can also fix these annoying warnings in ocl code and see real errors in the short log :)
But I think at this moment we should proceed with the first option.

Reply:
@vladimir-paramuzov OK. I updated the PR as you said. I think this is more developer friendly :)